src/pypromice/qc/github_data_issues.py

```diff
@@ -176,13 +176,11 @@ def adjustData(ds,
         # removing potential time shifts from the adjustment list
         adj_info = adj_info.loc[adj_info.adjust_function != "time_shift", :]
 
-        # if t1 is left empty, then adjustment is applied until the end of the file
-        adj_info.loc[adj_info.t0.isnull(), "t0"] = ds_out.time.values[0]
-        adj_info.loc[adj_info.t1.isnull(), "t1"] = ds_out.time.values[-1]
-        # making all timestamps timezone naive (compatibility with xarray)
-        adj_info.t0 = pd.to_datetime(adj_info.t0).dt.tz_localize(None)
-        adj_info.t1 = pd.to_datetime(adj_info.t1).dt.tz_localize(None)
-
+        # making sure that t0 and t1 columns are object dtype then replaceing nan with None
+        adj_info[['t0','t1']] = adj_info[['t0','t1']].astype(object)
+        adj_info.loc[adj_info.t1.isnull()|(adj_info.t1==''), "t1"] = None      
+        adj_info.loc[adj_info.t0.isnull()|(adj_info.t0==''), "t0"] = None
```

Comment:
`t0` and `t1` are already checked in line 219:222 except for the empty string and `np.nan`
Consider instead

```python
                if isinstance(t0, str) and t0 != '':
                    t0 = pd.to_datetime(t0, utc=True).tz_localize(None)
                else:
                    t0 = None
                if isinstance(t1, str) and t1 != '':
                    t1 = pd.to_datetime(t1, utc=True).tz_localize(None)
                else:
                    t1 = None
```

Reply:
Not a big fan of if/else in for loop. I'd rather prepare the `adj_info` table before looping through it.
Now I see that I have used part of your suggestion l.219-222 so I admit it is currently a bit hybrid.